src/workerd/api/container.c++

```diff
@@ -48,24 +48,38 @@ void Container::start(jsg::Lock& js, jsg::Optional<StartupOptions> maybeOptions)
   }
 
   IoContext::current().addTask(req.sendIgnoringResult());
+  IoContext::current().addTask(rpcClient->monitorRequest(capnp::MessageSize{4, 0})
+                                   .sendIgnoringResult()
+                                   .then([this, self = JSG_THIS]() { running = false; },
+                                       [this, self = JSG_THIS](kj::Exception&& error) {
+    running = false;
+
+    if (!monitoring) {
+      IoContext::current().abort(kj::mv(error));
+    }
+  }));
 
   running = true;
 }
 
 jsg::Promise<void> Container::monitor(jsg::Lock& js) {
   JSG_REQUIRE(running, Error, "monitor() cannot be called on a container that is not running.");
 
+  monitoring = true;
+
   return IoContext::current()
       .awaitIo(js, rpcClient->monitorRequest(capnp::MessageSize{4, 0}).sendIgnoringResult())
```

Comment:
I think instead of making a second monitor request, we should piggy-back on the existing background request, i.e. return a copy of the same promise.

Reply:
kj::Promise<void> doesn't have a copy constructor. Any other suggestions?

Reply:
Promises (kj::Promise) are not copyable. Am I missing something?